Formula/sampler.rb

```diff
@@ -0,0 +1,21 @@
+class Sampler < Formula
+  desc "Tool for shell commands execution, visualization and alerting"
+  homepage "https://sampler.dev"
+  url "https://github.com/sqshq/sampler/archive/v1.1.0.tar.gz"
+  sha256 "8b60bc5c0f94ddd4291abc2b89c1792da424fa590733932871f7b5e07e7587f9"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/sqshq/sampler"
+    dir.install Dir["*"]
+    cd dir do
+      system "go", "build", "-o", bin/"sampler"
+    end
+  end
+
+  test do
+    system "#{bin}/sampler", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev Sampler is a UI application, it does not produce any output. Here is a full list of available options:

```
Application Options:
  -c, --config=  Path to YAML config file
  -e, --env=     Specify name=value variable to use in script placeholder as $name. This flag takes precedence over the same name variables, specified in config yml
  -v, --version  Print version
```

Here is an example of a similar UI-only application, which you have in Homebrew-core:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/ctop.rb

It does the same `system "#{bin}/ctop", "-v"` test

Reply:
If you'd like, I can change the test to check that app fails when config file is not specified:

```ruby
  test do
    assert_includes "specify config file", shell_output("#{bin}/sampler")
  end
```

Reply:
done

Reply:
👍